packages/react/src/core/FileTree.tsx

```diff
@@ -183,6 +183,7 @@ interface FileProps {
 }
 
 function File({ file: { depth, name }, onClick, selected }: FileProps) {
+  const fileIcon = name ? getFileTreeIcon(name) : '';
```

Comment:
Overall looks good! 💯 

Couple of improvement ideas:

- Let's create a `getFileExtension` function that can be called with this `name`. 
  ```js
  function getFileExtension(filename: string) {
    const parts = filename.split('.');

    // drop the filename if there's no extension
    parts.shift();

    const extension = parts.at(-1) || '';

    return extension
  }
  ```

- Create a `extensionsToIcons` `Map` that holds all the extension-to-icon mappings. This should be created top-level somewhere between `interface Props` and `export function FileTree`.
  ```js
  const extensionsToIcons = new Map([
    ['ts', 'i-ph-file-ts-duotone'],

    ['cjs', 'i-ph-file-js-duotone'],
    ['mjs', 'i-ph-file-js-duotone'],
    ['js', 'i-ph-file-js-duotone'],

    ['html', 'i-ph-file-html-duotone'],

    ... and so on
  ]);
  ```

- Let's replace this line 186 with following:
  ```diff
  - const fileIcon = name ? getFileTreeIcon(name) : '';
  + const extension = getFileExtension(name);
  + const fileIcon = extensionsToIcons.get(extension) || 'i-ph-file-duotone';
  ```

Reply:
Could you also create some examples files in `packages/template/src/content/tutorial/1-basics/1-introduction/2-foo/_files/src` so that we can easily test all of these extensions when opening `http://localhost:4321/1-basics/custom-chapter-slug/foo`.

Reply:
Hi there! I made the necessary changes to the code, but it seems the GitHub actions failed at first because I missed adding a newline before the return. I went ahead and edited the file directly on GitHub, but it's still not passing the actions. Could you kindly take a look at the code and let me know what might be wrong? 😅

Reply:
The changes look good! I'll check rest of this PR next week, thanks!